File: features/discovery/search/src/main/java/com/tokopedia/search/di/module/FingerprintInterceptorModule.java
package com.tokopedia.search.di.module;

import android.content.Context;

import com.tokopedia.abstraction.common.di.qualifier.ApplicationContext;
import com.tokopedia.discovery.newdiscovery.di.scope.SearchScope;
import com.tokopedia.network.NetworkRouter;
import com.tokopedia.network.interceptor.FingerprintInterceptor;
import com.tokopedia.user.session.UserSessionInterface;

import dagger.Module;
import dagger.Provides;

@SearchScope
@Module
public class FingerprintInterceptorModule {

    @SearchScope
    @Provides
    public FingerprintInterceptor provideFingerprintInterceptor(@ApplicationContext Context context, UserSessionInterface userSession) {
        return new FingerprintInterceptor((NetworkRouter) context, userSession);
    }
}

File: features/discovery/search/src/main/java/com/tokopedia/search/di/module/FingerprintInterceptorModule.kt
package com.tokopedia.search.di.module

import android.content.Context
import com.tokopedia.abstraction.common.di.qualifier.ApplicationContext
import com.tokopedia.search.di.scope.SearchScope
import dagger.Provides
import com.tokopedia.user.session.UserSessionInterface
import com.tokopedia.network.NetworkRouter
import com.tokopedia.network.interceptor.FingerprintInterceptor
import dagger.Module

@Module
class FingerprintInterceptorModule {
    @SearchScope
    @Provides
    fun provideFingerprintInterceptor(@ApplicationContext context: Context?, userSession: UserSessionInterface?): FingerprintInterceptor {
        return FingerprintInterceptor(context as NetworkRouter?, userSession)
    }
}